### PR TITLE
Feature: Add converte Array of Object

### PR DESCRIPTION
**Pull request by xPolo360**
Hello,

Thank you for your library.
I allow myself a small contribution because I noticed that your library does not take into account the case for the original was given an array of javacript object.

example:
```javascript
toCamel(
  [ // <== ARRAY
    { fooBar: 'fooBar', barBaz: 1 },
    { fooBar: 'foo', quxBar: 'bar' }
  ]
);
// yields
// { <== OBJECT
//    { fooBar: 'fooBar', barBaz: 1 },
//    { fooBar: 'foo', quxBar: 'bar' }
// }
```

I allows some optimization of typing and code.

Regards.

### CODE REVIEW

**Comment by craftgear** [comment]
Hi,

Thanks for your contribution!

As you pointed out, the implementation for an `Array` of `Object`s are omitted.
And it's intentional.

Because you can just map `toCamel` or `toSnake` over an `Array` and the job gets done.
Like so:

```
  const camelCased = [
    { fooBar: 'fooBar', barBaz: 1 },
    { fooBar: 'foo', quxBar: 'bar' }
  ];

  var snaked = camelCased.map(toSnake);
```

I prefer handling this with innate JavaScript functions to covering it in my small library.

I'll include this use case in specs and `README.md`.

Thanks again.

**Comment by xPolo360** [comment]
Sure it's a way to do which I had not thought.

The idea was for me to put in my helper fetch the'toCamel to convert dice the arrival of my API.

And in this case my api sometimes returns objects and sometimes arrays depending on whether I want to retrieve a resource or a collection.
And in this case I do not have to ask myself the question of knowing if it is an object or an array your library takes care of it.

But I understand your point of view.

Afterwards you can keep the typings that I suggest to you because it allows to keep the type of the return according to the typings of input.

For autocompletion it's much more practical;)

**Comment by craftgear** [comment]
Sweet! Let me take a look at  it.

File: src/index.spec.ts [hunk 1]
```diff
@@ -51,3 +51,21 @@ describe('convert object properties', () => {
...
+    { foo_bar: 'foo', qux_bar: 'bar' }
+  ];
+  it('from camelCase to snake_case', () => {
+    expect(toCamel(snakeCased)).toEqual(camelCased);
+  });
+
+  it('from snake_case to camelCase', () => {
+    expect(toSnake(camelCased)).toEqual(snakeCased);
+  });
+});
```
Comment (craftgear) [inline src/index.spec.ts:71]: I think you can safely delete these specs.

File: src/index.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ const propertyNameConverter = (converterFn: (s: string) => string) => (data: Any
       };
     }, {});
   };
-  return recursive(data);
+  return recursive({ key: data }).key;
```
Comment (craftgear) [inline src/index.ts:36]: Where does this `key` come from?

**Review by craftgear — CHANGES_REQUESTED** [review]

**Comment by craftgear** [comment]
It's stale, closing this.

**Pull request closed without merging**